src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -1127,6 +1127,155 @@ TIP: Since it runs a graphical application, the task will never exit and run for
 you want to check manually some results via the graphical application and then exit the application.
 The task will then terminate normally.
 
+== Workflow Storage And Versioning
+
+The Workflow Catalog provides the storage and versioning of workflows inside Buckets. It also features a searching functionality using a simple query language to fetch workflows based on particular parameters. The Workflow Catalog is a RESTful service. A http://swagger.io/[Swagger] interface is provided to understand and interact with the service.
+
+=== Bucket concept
+
+A Bucket is a collection of Workflows that can be shared between users. The default settings provides the following Buckets:
+
+* Catalog
+* Templates (set of Workflows samples publicly readable)
+
+Listing the catalog for existing Buckets is done using the following HTTP request:
+
+    GET http://localhost:8080/workflow-catalog/buckets
+
+It is also possible to list all Workflows inside a particular Bucket using its `bucket_id`:
+
+    GET http://localhost:8080/workflow-catalog/buckets/{bucket_id}/workflows
+
+Then we can fetch a particular Workflow based on its `workflow_id`:
+
+    GET http://localhost:8080/workflow-catalog/buckets/{bucket_id}/workflows/{workflow_id}
+
+=== Workflow versioning
+
+The Workflow Catalog provides versioning of workflows. You can list all the revisions of a particular Workflow using the HTTP request:
+
+    GET http://localhost:8080/workflow-catalog/buckets/{bucket_id}/workflows/{workflow_id}/revisions
+
+You can fetch a Workflow detailed information using its `workflow_id` and `revision_id`:
+
+    GET http://localhost:8080/workflow-catalog/buckets/{bucket_id}/workflows/{workflow_id}/revisions/{revision_id}
+
+NOTE: By default, the catalog will always return the latest version of the workflows.
+
+A Workflow Revision is created from a ProActive XML <<_glossary_workflow,*Workflow*>> definition.
+Beside from the _id_ and the _name_ of the Workflow, a few more information are indexed from the XML content such as _Variables_ and _Generic Information_.
+
+More detailed information on how to create or modify Workflows and Buckets are available in the link:../admin/ProActiveAdminGuide.html#_workflow_catalog[Admin guide].
+
+
+=== Searching for workflows
+
+You can search or filter Workflows and Workflow Revisions by means of Workflow Catalog Query Language (<<_wcql,*WCQL*>>).
+
+Search on the latest versions of Workflows contained in a particular Bucket:
+
+    http://localhost:8080/workflow-catalog/{bucket_id}/workflows
+
+Search through all versions of a particular Workflow:
+
+    http://localhost:8080/workflow-catalog/{bucket_id}/workflows/{workflow_id}/revisions
+
```

Comment:
We could maybe add a note about query parameters for pagination (and optionally sorting)?